hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/management/JpaDeploymentManagement.java

```diff
@@ -187,14 +187,20 @@ public JpaDeploymentManagement(final EntityManager entityManager, final ActionRe
     @Transactional(isolation = Isolation.READ_COMMITTED)
     public List<DistributionSetAssignmentResult> offlineAssignedDistributionSets(
             final Collection<Entry<String, Long>> assignments) {
+            return offlineAssignedDistributionSets(assignments,tenantAware.getCurrentUsername());
+    }
+
+    @Override
+    public List<DistributionSetAssignmentResult> offlineAssignedDistributionSets(
+            Collection<Entry<String, Long>> assignments, String initiatedBy) {
         final Collection<Entry<String, Long>> distinctAssignments = assignments.stream().distinct().toList();
         enforceMaxAssignmentsPerRequest(distinctAssignments.size());
 
         final List<DeploymentRequest> deploymentRequests = distinctAssignments.stream()
                 .map(entry -> DeploymentManagement.deploymentRequest(entry.getKey(), entry.getValue()).build())
                 .toList();
 
-        return assignDistributionSets(tenantAware.getCurrentUsername(), deploymentRequests, null,
+        return assignDistributionSets(Objects.requireNonNull(initiatedBy), deploymentRequests, null,
```

Comment:
I'm not sure that here we shall set controllerId as lastModifiedBy
AbstractJpaBaseEntity.setLastModifiedBy explicitly check if it is controller and skips it
maybe null shall be set